registry/quilt_server/views.py

```diff
@@ -886,6 +886,11 @@ def package_preview(owner, package_name, package_hash):
     instance = _get_instance(g.auth, owner, package_name, package_hash)
     assert isinstance(instance.contents, RootNode)
 
+    package = _get_package(g.auth, owner, package_name)
```

Comment:
`instance.package`

Reply:
Also, see the `packages` query in `search()`; you can try to change `_get_instance` to return the instance as well as `is_team` and `is_public`